### PR TITLE
Improving code efficiency by errors from cppcheck

### PR DESCRIPTION
**Pull request by Surajjalpun2002**
Raising PR for issue #1346 

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: avogadro/core/angleiterator.cpp
Comment (Surajjalpun2002) [inline avogadro/core/angleiterator.cpp:1]: Also used clang-format on the 2 files.

**Comment by ghutchis** [comment]
I'll take a look at the clang-format check on Thursday.

File: avogadro/core/variant.h
Comment (Surajjalpun2002) [inline avogadro/core/variant.h:1]: this commit somehow has message from the previous commit, but changes in it are for the just next commit about removing unused private function

File: avogadro/core/array.h [hunk 1]
```diff
@@ -118,19 +121,18 @@ class Array
...
-
-  /** Copy constructor, note the copy made of the internal data of other. */
-  Array(const Array& other)
   {
-    other.d->reref();
-    d = other.d;
   }
 
+  /** Copy constructor, note the copy made of the internal data of other. */
+  Array(const Array& other) : d(new Container(*other.d)) {}
```
Comment (ghutchis) [inline avogadro/core/array.h:133]: Yeah, we don't want to do this. The Array class is copy-on-write. Please remove and/or tell cppcheck to ignore this.

Comment (Surajjalpun2002) [reply avogadro/core/array.h:133]: ok

**Comment by Surajjalpun2002** [comment]
Any tips?

**Comment by ghutchis** [comment]
If you click "details" on any of the tests, you should be able to see the build report:
```
/Users/runner/actions-runner/_work/avogadrolibs/avogadrolibs/avogadrolibs/avogadro/core/variant.h:133:3: error: expected member name or ';' after declaration specifiers
  private : Type m_type;
  ^
```

(Then lots more errors)

**Comment by ghutchis** [comment]
Please, let's stick to small changes. I'd rather have multiple PR with individual small changes and merge them, then have you keep pushing to this one. I don't know when you're "done" so I can review and give comments and test. (Beyond the unit tests, it's important to do functional testing because small changes can still yield big changes, e.g. in `Array`)

**Comment by Surajjalpun2002** [comment]
image

**Comment by Surajjalpun2002** [comment]
Should i close this PR as well and open a new one, as most of the changes ive made have been causing errors and i believe there is only one small commit thats reasonable in this whole pr.

**Comment by ghutchis** [comment]
You're certainly not a hassle, but definitely submitting small changes as separate PR for testing would be appreciated.

Understand that `cppcheck` and other static analyzers are not perfect. They flag plenty of false positives. They do also offer useful suggestions.

For example the `cube.cpp` issues you note above are probably useful to fix.

**Pull request closed without merging**